src/opm/input/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -799,51 +831,127 @@ inline void keywordB( SummaryConfig::keyword_list& list,
-    std::vector<int> regions;
+    return region_name;
+}
+
+inline void keywordR2R_unsupported(const DeckKeyword&  keyword,
+                                   const ParseContext& parseContext,
+                                   ErrorGuard&         errors)
+{
+    const auto msg_fmt = std::string {
+        "Region to region summary keyword {keyword} is ignored\n"
+        "In {file} line {line}"
+    };
+
+    parseContext.handleError(ParseContext::SUMMARY_UNHANDLED_KEYWORD,
+                             msg_fmt, keyword.location(), errors);
+}
+
+inline void keywordR2R(const DeckKeyword&           keyword,
+                       const FieldPropsManager&     field_props,
+                       const ParseContext&          parseContext,
+                       ErrorGuard&                  errors,
+                       SummaryConfigContext&        context,
+                       SummaryConfig::keyword_list& list)
+{
+    if (is_unsupported_region_to_region(keyword.name())) {
+        keywordR2R_unsupported(keyword, parseContext, errors);
+    }
+
+    if (is_udq(keyword.name())) {
+        throw std::invalid_argument {
+            "Inter-Region quantity '"
+           + keyword.name() + "' "
+           + "cannot be a user-defined quantity"
+        };
+    }
+
+    const auto region_name = establishRegionContext(keyword, field_props,
+                                                    parseContext, errors,
+                                                    context);
+
+    if (region_name.empty()) {
+        return;
+    }
+
+    auto param = SummaryConfigNode {
+        keyword.name(), SummaryConfigNode::Category::Region, keyword.location()
+    }
+    .parameterType(parseKeywordType(keyword.name()))
+    .fip_region(region_name)
+    .isUserDefined(false);
+
+    // Expected format:
+    //
+    //   ROFT
+    //     1 2 /
+    //     1 4 /
+    //   /
+    for (const auto& record : keyword) {
+        // We *intentionally* record/use one-based region IDs here.
+        const auto r1 = record.getItem("REGION1").get<int>(0);
+        const auto r2 = record.getItem("REGION2").get<int>(0);
+
+        list.push_back(param.number(EclIO::combineSummaryNumbers(r1, r2)));
+    }
+}
 
 
+inline void keywordR(SummaryConfig::keyword_list& list,
+                     SummaryConfigContext&        context,
+                     const DeckKeyword&           deck_keyword,
+                     const Schedule&              schedule,
+                     const FieldPropsManager&     field_props,
+                     const ParseContext&          parseContext,
+                     ErrorGuard&                  errors)
+{
+    const auto keyword = deck_keyword.name();
+    if (is_region_to_region(keyword)) {
+        keywordR2R(deck_keyword, field_props, parseContext, errors, context, list);
+        return;
+    }
+
+    const auto region_name = establishRegionContext(deck_keyword, field_props,
```

Comment:
What about `std::optional<std::string>`?

Reply:
> How about `std::optional<std::string>`?

Yes, that's a good idea.  I've pushed an update to that effect.